pkg/network/prober/prober.go

```diff
@@ -34,6 +34,9 @@ type Preparer func(r *http.Request) *http.Request
 // Verifier is a way for the caller to validate the HTTP response after it comes back.
 type Verifier func(r *http.Response, b []byte) (bool, error)
 
+// Exception is a way for the caller to validate the HTTP response after the verifier's validation.
```

Comment:
this is more like "PostValidate"
Also does it need body? I struggle to see post validation needing body (all that can be verified before).

Reply:
Right, implement this as a Verifier.
Something like
func ExpectsStatusCodes(statusCodes []int) (bool, error) {
  ...
}

And add use it everywhere Probe is used. Or if no ExpectsStatusCodes is provided, insert a EXpectsstatusCodes([]{200}) for backward compatibility.

Reply:
Thank you. I updated it with the former one. I wanted to do the latter, but it would not be possible with clean code. I had an idea to use `runtime.FuncForPC()` which can get the function name, but it is too ugly I think.